res/controllers/lib/mididispatcher.mjs

```diff
@@ -0,0 +1,24 @@
+export class MidiDispatcher {
+    constructor(noteOff) {
+        this.noteOff = noteOff;
+        this.inputMap = new Map();
+    }
+    registerInputCallback(midiBytes, callback) {
+        // JavaScript is broken and believes [1,2] === [1,2] is false, so
+        // JSONify the Array to make it usable as a Map key.
+        const key = JSON.stringify(midiBytes);
+        this.inputMap.set(key, callback);
+        if (this.noteOff === true && ((midiBytes[0] & 0xF0) === 0x90)) {
```

Comment:
Maybe it's obvious but can you add a comment that explains what this code does? 

Are there controllers where it would make sense to set this per registerInputCallback call?

Reply:
I added a comment: f1a1b01

Reply:
Thanks.